packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -1010,16 +1013,15 @@ class _InkResponseState extends State<_InkResponseStateWidget>
       updateKeepAlive();
     }
 
-    switch (type) {
-      case _HighlightType.pressed:
-        statesController.update(WidgetState.pressed, value);
-      case _HighlightType.hover:
-        if (callOnHover) {
-          statesController.update(WidgetState.hovered, value);
-        }
-      case _HighlightType.focus:
-        // see handleFocusUpdate()
-        break;
+    // Update the appropriate WidgetState for this highlight type.
+    if (type == _HighlightType.pressed) {
+      statesController.update(WidgetState.pressed, value);
+    } else if (type == _HighlightType.hover) {
+      if (callOnHover) {
+        statesController.update(WidgetState.hovered, value);
+      }
+    } else {
+      // _HighlightType.focus: handled in handleFocusUpdate().
```

Comment:
This part of change seems unnecessary. The two versions are equivalent, except that `switch` also checks exhaustiveness.